lib/block-supports/duotone.php

```diff
@@ -433,61 +436,51 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
-	// 1. Array of colors - e.g. array('#000000', '#ffffff').
-	// 2. Slug of an existing Duotone preset - e.g. 'green-blue'.
-	// 3. The string 'unset' - indicates explicitly "no Duotone"..
-	$duotone_attr = $block['attrs']['style']['color']['duotone'];
-
-	$is_duotone_colors_array = is_array( $duotone_attr );
-	$is_duotone_unset        = 'unset' === $duotone_attr;
-	$is_duotone_preset       = ! $is_duotone_colors_array && ! $is_duotone_unset;
+	if ( $has_custom_duotone ) {
+		// Handle custom values for duotone attribute.
 
-	if ( $is_duotone_preset ) {
-		$filter_preset = array(
-			'slug' => $duotone_attr,
-		);
-
-		// Utilise existing CSS custom property.
-		$filter_property = "var(--wp--preset--duotone--$duotone_attr)";
-	} else {
-		// Handle when Duotone is either:
-		// - "unset"
-		// - an array of colors.
+		// 1. Array of colors - e.g. array('#000000', '#ffffff').
+		// 2. The string 'unset' - indicates explicitly "override the theme duotone with `filter: none !important`".
+		$duotone_attr = $block['attrs']['style']['color']['duotone'];
 
 		// Build a unique slug for the filter based on the array of colors.
-		$filter_key    = $is_duotone_colors_array ? implode( '-', $duotone_attr ) : $duotone_attr;
-		$filter_preset = array(
+		$filter_key  = is_array( $duotone_attr ) ? implode( '-', $duotone_attr ) : $duotone_attr;
+		$filter_data = array(
 			'slug'   => wp_unique_id( sanitize_key( $filter_key . '-' ) ),
 			'colors' => $duotone_attr, // required for building the SVG with gutenberg_get_duotone_filter_svg.
 		);
 
-		// Build a customised CSS filter property for unique slug.
-		$filter_property = $is_duotone_unset ? 'none' : gutenberg_get_duotone_filter_property( $filter_preset );
+		// Build a customized CSS filter property for unique slug.
+		$filter_property = gutenberg_get_duotone_filter_property( $filter_data );
+	} elseif ( $has_preset_duotone ) {
+		// Handle preset values for duotone iff the preset is set and there is no custom duotone.
+
+		// 1. Slug of an existing Duotone preset - e.g. 'green-blue'.
+		$duotone_attr = $block['attrs']['duotone'];
+		$filter_data    = array(
+			'slug' => $duotone_attr,
+		);
+
+		// Utilize existing CSS custom property.
+		$filter_property = "var(--wp--preset--duotone--$duotone_attr)";
 	}
 
 	// - Applied as a class attribute to the block wrapper.
 	// - Used as a selector to apply the filter to the block.
-	$filter_id = gutenberg_get_duotone_filter_id( $filter_preset );
+	$filter_id = gutenberg_get_duotone_filter_id( $filter_data );
 
 	// Build the CSS selectors to which the filter will be applied.
-	// Todo - encapsulate this in a function.
-	$scope     = '.' . $filter_id;
-	$selectors = explode( ',', $duotone_support );
-	$scoped    = array();
-	foreach ( $selectors as $sel ) {
-		$scoped[] = $scope . ' ' . trim( $sel );
-	}
-	$selector = implode( ', ', $scoped );
+	$selector = WP_Theme_JSON_Gutenberg::scope_selector( '.' . $filter_id, $duotone_support );
```

Comment:
I forget—when doing backports, renaming things like `WP_Theme_JSON_Gutenberg` to `WP_Theme_JSON` is part of the normal process, right? Or will this need special attention?